api/client/client.go

```diff
@@ -624,14 +619,10 @@ func (c *Client) DeleteAllKubeServices(ctx context.Context) error {
 }
 
 // GetDatabaseServers returns all registered database proxy servers.
-func (c *Client) GetDatabaseServers(ctx context.Context, namespace string, opts ...types.MarshalOption) ([]types.DatabaseServer, error) {
-	cfg, err := types.CollectOptions(opts)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
+func (c *Client) GetDatabaseServers(ctx context.Context, namespace string, skipValidation bool) ([]types.DatabaseServer, error) {
```

Comment:
We wrap all of the proto endpoints like this to hide implementation details, and make it easier to use. The user should not even need to we are using gRPC. Currently it looks like 4 or 5 other endpoints break this pattern, and I will need to go refactor those to use non proto-types.

Reply:
Using protobuf is not only for RPC - it is a convenient data description format and as such I see no reason to avoid using it even in client facing API unless exposing specific request attributes is undesirable.

Reply:
I see what you're saying, it could be that this is an anti-pattern. However if we want the client to take proto request types in the functions, there is no reason to define any of the functions in `client.go` (except maybe to wrap errors). We could simply embed `proto.AuthServiceClient` into `Client`.

However, we do have several functions that we want to take non `proto` types top enable the use of helper methods, such as `CreateUser(ctx, types.User)`. The pattern for using this would be:
```go
user := types.NewUser(...)
client.CreateUser(ctx, user)
```

If we mix proto and `types` types into our client, I think it could become confusing, and so in my eyes consistency is key here. I'd rather have all the endpoints take only `types` types and built in types.

I had raised your idea to @russjones a few weeks ago, but he said we should keep the client the way it is for reasons similar to what I've shared. I can definitely see benefits to both approaches though.

Reply:
Ok - it would nice to have an agreement on that to follow consistently since I can relate to this in the context of my other [PR](https://github.com/gravitational/teleport/pull/5276)